command/seal_migration_test.go

```diff
@@ -1,540 +1,352 @@
-	physInmem "github.com/hashicorp/vault/sdk/physical/inmem"
+	"github.com/hashicorp/vault/sdk/logical"
 	"github.com/hashicorp/vault/vault"
 	"github.com/hashicorp/vault/vault/seal"
 )
 
-func TestSealMigrationAutoToShamir(t *testing.T) {
-	logger := logging.NewVaultLogger(hclog.Trace).Named(t.Name())
-	phys, err := physInmem.NewInmem(nil, logger)
-	if err != nil {
+type transitSealServer struct {
+	*vault.TestCluster
+}
+
+func newTransitSealServer(t *testing.T) *transitSealServer {
+	conf := &vault.CoreConfig{
+		LogicalBackends: map[string]logical.Factory{
+			"transit": transit.Factory,
+		},
+	}
+	opts := &vault.TestClusterOptions{
+		NumCores:    1,
+		HandlerFunc: vaulthttp.Handler,
+		Logger:      logging.NewVaultLogger(hclog.Trace).Named(t.Name()).Named("transit"),
+	}
+	teststorage.InmemBackendSetup(conf, opts)
+	cluster := vault.NewTestCluster(t, conf, opts)
+	cluster.Start()
+
+	if err := cluster.Cores[0].Client.Sys().Mount("transit", &api.MountInput{
+		Type: "transit",
+	}); err != nil {
 		t.Fatal(err)
 	}
-	haPhys, err := physInmem.NewInmemHA(nil, logger)
-	if err != nil {
+
+	return &transitSealServer{cluster}
+}
+
+func (tss *transitSealServer) makeKey(t *testing.T, key string) {
+	client := tss.Cores[0].Client
+	// Create default aesgcm key
+	if _, err := client.Logical().Write(path.Join("transit", "keys", key), nil); err != nil {
+		t.Fatal(err)
+	}
+	if _, err := client.Logical().Write(path.Join("transit", "keys", key, "config"), map[string]interface{}{
+		"deletion_allowed": true,
+	}); err != nil {
 		t.Fatal(err)
 	}
-	autoSeal := vault.NewAutoSeal(seal.NewTestSeal(nil))
-	cluster := vault.NewTestCluster(t, &vault.CoreConfig{
-		Seal:            autoSeal,
-		Physical:        phys,
-		HAPhysical:      haPhys.(physical.HABackend),
+}
+
+func (tss *transitSealServer) makeSeal(t *testing.T, key string) vault.Seal {
+	client := tss.Cores[0].Client
+	wrapperConfig := map[string]string{
+		"address":     client.Address(),
+		"token":       client.Token(),
+		"mount_path":  "transit",
+		"key_name":    key,
+		"tls_ca_cert": tss.CACertPEMFile,
+	}
+	transitSeal, _, err := commandseal.GetTransitKMSFunc(nil, wrapperConfig)
+	if err != nil {
+		t.Fatalf("error setting wrapper config: %v", err)
+	}
+
+	return vault.NewAutoSeal(&seal.Access{
+		Wrapper: transitSeal,
+	})
+}
+
+func verifyBarrierConfig(t *testing.T, cfg *vault.SealConfig, sealType string, shares, threshold, stored int) {
+	t.Helper()
+	if cfg.Type != sealType {
+		t.Fatalf("bad seal config: %#v, expected type=%q", cfg, sealType)
+	}
+	if cfg.SecretShares != shares {
+		t.Fatalf("bad seal config: %#v, expected SecretShares=%d", cfg, shares)
+	}
+	if cfg.SecretThreshold != threshold {
+		t.Fatalf("bad seal config: %#v, expected SecretThreshold=%d", cfg, threshold)
+	}
+	if cfg.StoredShares != stored {
+		t.Fatalf("bad seal config: %#v, expected StoredShares=%d", cfg, stored)
+	}
+}
+
+func TestSealMigrationShamirToAuto(t *testing.T) {
```

Comment:
Small nit, can we rename these tests to `TestSealMigration_*` (e.g. `TestSealMigration_ShamirToAuto`)?

Reply:
Certainly.